de.metas.business.rest-api/src/main/java/de/metas/rest_api/invoicecandidates/request/JsonInvoiceCandCreateRequest.java

```diff
@@ -0,0 +1,91 @@
+package de.metas.rest_api.invoicecandidates.request;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Singular;
+import lombok.Value;
+
+/*
+ * #%L
+ * de.metas.business.rest-api
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+@Value
+public final class JsonInvoiceCandCreateRequest
+{
+	@ApiModelProperty( //
+			allowEmptyValue = false, //
+			value = "These are the invoices json objects</code>.\n")
+	List<JsonInvoiceCandidate> jsonInvoices;
+
+	@ApiModelProperty(value = "Date of the invoice.")
+	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+	@JsonInclude(Include.NON_NULL)
+	private LocalDate dateInvoiced;
+
+	@ApiModelProperty(value = "Accounting date.")
+	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+	@JsonInclude(Include.NON_NULL)
+	private LocalDate dateAcct;
+
+	@ApiModelProperty(value = "External reference (document number) on a remote system. Not neccesarily unique, but but the external user will want to filter recrods using it")
+	private String poReference;
+
+	@ApiModelProperty(required = true, value = "Ignore invoice schedule.")
```

Comment:
I took C_Invoice_Candidate_EnqueueSelectionForInvoicing as an example as you suggested and thought the same invoicing params are needed in our case as well. I've seen that they are used in the process of **generate invoices**. To be honest, I am not sure how to find out if we need all of them or not.I have found this parameter to be used in here:
![image](https://user-images.githubusercontent.com/50820854/67359390-ca8c2a00-f56b-11e9-855a-076cf6a134da.png)
 Instead of **Ignore invoice schedule** should I write **If this is set, the invoice candidate is not eligible for enqueueing** ? I don't have any other idea.

Reply:
in such cases, please ask in the team chat daniela and very probably also other will know.
(it's about overriding a setting that that e.g. invoices shall be created only on the 1st of the respective next month. 
With this parameter you ignore taht setting, if you need the invoice right now)

Reply:
should I ask if we need this param in this implementation or you mean that I should ask what it does?

Reply:
please ask what it does. i don't see why we should keep any particular of those parameters away from the API-users, if the webui-users have it too

Reply:
done